src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java

```diff
@@ -0,0 +1,203 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.config;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Maps.EntryTransformer;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.yaml.snakeyaml.Yaml;
+
+@AutoValue
+public abstract class PackageMetadataConfig {
+
+  @AutoValue
+  public abstract static class VersionBound {
+    public abstract String lower();
+
+    @Nullable
+    public abstract String upper();
+  }
+
+  protected abstract Map<String, VersionBound> gaxVersion();
+
+  protected abstract Map<String, VersionBound> protoVersion();
+
+  protected abstract Map<String, VersionBound> commonProtosVersion();
+
+  protected abstract Map<String, VersionBound> packageVersion();
+
+  protected abstract Map<String, String> packageName();
+
+  /** The version of GAX that this package depends on. Configured per language. */
+  public VersionBound gaxVersion(String language) {
```

Comment:
I'd prefer not to pass around language identifiers as plain strings - lots of room for odd variations & mistakes, and it is an "almost" closed set.